src/math/cbrtf.rs

```diff
@@ -12,65 +11,52 @@
  * software is freely granted, provided that this notice
  * is preserved.
  * ====================================================
+ *
  */
-/* cbrtf(x)
- * Return cube root of x
- */
 
-use core::f32;
+use super::fdlibm::{FLT_UWORD_IS_FINITE, FLT_UWORD_IS_SUBNORMAL, FLT_UWORD_IS_ZERO};
+
+const B1: u32 = 709_958_130; /* B1 = (84+2/3-0.03306235651)*2**23 */
+const B2: u32 = 642_849_266; /* B2 = (76+2/3-0.03306235651)*2**23 */
 
-const B1: u32 = 709958130; /* B1 = (127-127.0/3-0.03306235651)*2**23 */
-const B2: u32 = 642849266; /* B2 = (127-127.0/3-24/3-0.03306235651)*2**23 */
+const C: f32 = 5.428_571_701_0e-01; /* 19/35        = 0x3f0a_f8b0 */
+const D: f32 = -7.053_061_127_7e-01; /* -864/1225   = 0xbf34_8ef1 */
+const E: f32 = 1.414_285_659_8e+00; /* 99/70        = 0x3fb5_0750 */
+const F: f32 = 1.607_142_806_1e+00; /* 45/28        = 0x3fcd_b6db */
+const G: f32 = 3.571_428_656_6e-01; /* 5/14         = 0x3eb6_db6e */
 
 /// Cube root (f32)
 ///
 /// Computes the cube root of the argument.
 #[inline]
 #[cfg_attr(all(test, assert_no_panic), no_panic::no_panic)]
-pub fn cbrtf(x: f32) -> f32 {
-    let x1p24 = f32::from_bits(0x4b800000); // 0x1p24f === 2 ^ 24
+pub extern "C" fn cbrtf(x: f32) -> f32 {
+    let hx: u32 = x.to_bits() & 0x7fff_ffff; /* |x| */
+    let sign: u32 = x.to_bits() & 0x8000_0000; /* sign = sign(x) */
 
-    let mut r: f64;
-    let mut t: f64;
-    let mut ui: u32 = x.to_bits();
-    let mut hx: u32 = ui & 0x7fffffff;
-
-    if hx >= 0x7f800000 {
-        /* cbrt(NaN,INF) is itself */
-        return x + x;
+    if !FLT_UWORD_IS_FINITE(hx) {
+        return x + x; /* cbrt(NaN,INF) is itself */
+    }
+    if FLT_UWORD_IS_ZERO(hx) {
+        return x; /* cbrt(0) is itself */
     }
 
+    let x = f32::from_bits(hx); /* x <- |x| */
     /* rough cbrt to 5 bits */
-    if hx < 0x00800000 {
-        /* zero or subnormal? */
-        if hx == 0 {
-            return x; /* cbrt(+-0) is itself */
-        }
-        ui = (x * x1p24).to_bits();
-        hx = ui & 0x7fffffff;
-        hx = hx / 3 + B2;
+    let mut t: f32 = if FLT_UWORD_IS_SUBNORMAL(hx) {
+        /* subnormal number */
+        let t: f32 = f32::from_bits(0x4b80_0000);
```

Comment:
This can be a const.

Reply:
Done. But rustc would optimize it away as show in https://godbolt.org/z/tfeV0c.